File: core/src/test/java/de/qabel/core/accounting/CloseableHttpResponseStub.java
package de.qabel.core.accounting;

import org.apache.http.*;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.message.BasicHeader;
import org.apache.http.message.BasicStatusLine;
import org.apache.http.params.HttpParams;

import java.io.IOException;
import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

public class CloseableHttpResponseStub implements CloseableHttpResponse {
    public boolean closed;
    public StatusLine statusLine = new BasicStatusLine(new ProtocolVersion("HTTP", 1, 0), 200, "OK");
    public HttpEntity entity;

    @Override
    public void close() throws IOException {
        closed = true;
    }

    @Override
    public StatusLine getStatusLine() {
        return statusLine;
    }

    @Override
    public void setStatusLine(StatusLine statusline) {
        statusLine = statusline;
    }

    @Override
    public void setStatusLine(ProtocolVersion ver, int code) {
    }

    @Override
    public void setStatusLine(ProtocolVersion ver, int code, String reason) {

    }

    @Override
    public void setStatusCode(int code) throws IllegalStateException {
        statusLine = new BasicStatusLine(statusLine.getProtocolVersion(), code, "");
    }

    @Override
    public void setReasonPhrase(String reason) throws IllegalStateException {

    }

    @Override
    public HttpEntity getEntity() {
        return entity;
    }

    @Override
    public void setEntity(HttpEntity entity) {
        this.entity = entity;
    }

    @Override
    public Locale getLocale() {
        return null;
    }

    @Override
    public void setLocale(Locale loc) {

    }

    @Override
    public ProtocolVersion getProtocolVersion() {
        return null;
    }

    @Override
    public boolean containsHeader(String name) {
        return headers.containsKey(name);
    }

    @Override
    public Header[] getHeaders(String name) {
        return new Header[]{ new BasicHeader(name, headers.get(name)) };
    }

    @Override
    public Header getFirstHeader(String name) {
        return new BasicHeader(name, headers.get(name));
    }

    @Override
    public Header getLastHeader(String name) {
        return getFirstHeader(name);
    }

    @Override
    public Header[] getAllHeaders() {
        return new Header[0];
    }

    private Map<String, String> headers = new HashMap<>();

    @Override
    public void addHeader(Header header) {
        headers.put(header.getName(), header.getValue());
    }

    @Override
    public void addHeader(String name, String value) {
        headers.put(name, value);
    }

    @Override
    public void setHeader(Header header) {
        addHeader(header);
    }

    @Override
    public void setHeader(String name, String value) {
        addHeader(name, value);
    }

    @Override
    public void setHeaders(Header[] headers) {
        for (Header header : headers) {
            setHeader(header);
        }
    }

    @Override
    public void removeHeader(Header header) {
        headers.remove(header.getName());
    }

    @Override
    public void removeHeaders(String name) {
        headers.remove(name);
    }

    @Override
    public HeaderIterator headerIterator() {
        return null;
    }

    @Override
    public HeaderIterator headerIterator(String name) {
        return null;
    }

    @Override
    public HttpParams getParams() {
        return null;
    }

    @Override
    public void setParams(HttpParams params) {

    }
}

File: core/src/test/java/de/qabel/core/accounting/CloseableHttpResponseStub.kt
package de.qabel.core.accounting

import org.apache.http.*
import org.apache.http.client.methods.CloseableHttpResponse
import org.apache.http.message.BasicStatusLine
import org.apache.http.params.HttpParams

import java.io.IOException
import java.util.Locale

class CloseableHttpResponseStub : CloseableHttpResponse {
    override fun setEntity(entity: HttpEntity?) {
        this.privateEntity = entity
    }

    override fun getEntity(): HttpEntity {
        return this.privateEntity!!
    }

    var closed: Boolean = false
    private var statusLine: StatusLine = BasicStatusLine(ProtocolVersion("HTTP", 1, 0), 200, "OK")
    private var privateEntity: HttpEntity? = null

    @Throws(IOException::class)
    override fun close() {
        closed = true
    }

    override fun getStatusLine(): StatusLine {
        return statusLine
    }

    override fun setStatusLine(statusline: StatusLine) {
        statusLine = statusline
    }

    override fun setStatusLine(ver: ProtocolVersion, code: Int) {
    }

    override fun setStatusLine(ver: ProtocolVersion, code: Int, reason: String) {

    }

    @Throws(IllegalStateException::class)
    override fun setStatusCode(code: Int) {
        statusLine = BasicStatusLine(statusLine.protocolVersion, code, "")
    }

    @Throws(IllegalStateException::class)
    override fun setReasonPhrase(reason: String) {

    }

    override fun getLocale(): Locale? {
        return null
    }

    override fun setLocale(loc: Locale) {

    }

    override fun getProtocolVersion(): ProtocolVersion? {
        return null
    }

    override fun containsHeader(name: String): Boolean {
        return false
    }

    override fun getHeaders(name: String): Array<Header> {
        return arrayOfNulls<Int>(0) as Array<Header>
    }

    override fun getFirstHeader(name: String): Header? {
        return null
    }

    override fun getLastHeader(name: String): Header? {
        return null
    }

    override fun getAllHeaders(): Array<Header> {
        return arrayOfNulls<Int>(0) as Array<Header>
    }

    override fun addHeader(header: Header) {

    }

    override fun addHeader(name: String, value: String) {

    }

    override fun setHeader(header: Header) {

    }

    override fun setHeader(name: String, value: String) {

    }

    override fun setHeaders(headers: Array<Header>) {

    }

    override fun removeHeader(header: Header) {

    }

    override fun removeHeaders(name: String) {

    }

    override fun headerIterator(): HeaderIterator? {
        return null
    }

    override fun headerIterator(name: String): HeaderIterator? {
        return null
    }

    override fun getParams(): HttpParams? {
        return null
    }

    override fun setParams(params: HttpParams) {

    }
}
